platform/linuxbsd/x11/display_server_x11.cpp

```diff
@@ -821,12 +821,30 @@ Size2i DisplayServerX11::screen_get_size(int p_screen) const {
 	return _screen_get_rect(p_screen).size;
 }
 
+// A Handler to avoid crashing on non-fatal X errors by default.
+//
+// The original X11 error formatter `_XPrintDefaultError` is defined here:
+// https://gitlab.freedesktop.org/xorg/lib/libx11/-/blob/e45ca7b41dcd3ace7681d6897505f85d374640f2/src/XlibInt.c#L1322
+// It is not exposed through the API, accesses X11 internals,
+// and is much more complex, so this is a less complete simplified error X11 printer.
+int default_window_error_handler(Display *display, XErrorEvent *error) {
+	static char message[1024];
+	XGetErrorText(display, error->error_code, message, sizeof(message));
+
+	ERR_PRINT(vformat("Unhandled XServer error: %s"
+					  "\n   Major opcode of failed request: %d"
+					  "\n   Serial number of failed request: %d"
+					  "\n   Current serial number in output stream: %d",
+			String(message), error->request_code, error->minor_code, error->serial));
```

Comment:
```suggestion
			String::utf8(message), error->request_code, error->minor_code, error->serial));
```

According to docs: "The returned text is in the encoding of the current locale.", so it's likely UTF-8 in most cases, `String(x)` is for ASCII only.

Reply:
@mxnemu Are you able to do this change? This seems to be the only step left before merging.